### PR TITLE
Feat+error meaning

### PR DESCRIPTION
**Pull request by nojhan**
Basically displays the (guessed) meaning of the last error code (feature request #729).

Separated in two commits, so that you may allow a more or less granular level of details.

First commit: adds the core feature.
- Use a set of reasonably common error codes on POSIX systems
  (but is LP used out of a POSIX env?).
- Use the color than lp_error (namely LP_COLOR_ERR).
- Use a separated LP_ENABLE_ERROR_MEANING.
- Put in the default theme right after the error code.
- Code would be very much simpler/configurable with Bash 4's associative arrays,
  but here use a case switch with non-configurable error meanings.

Second commit: let the user decide the level of guess for error meaning.
- Adds LP_ENABLE_ERROR_MEANING_EXTENDED to allow for a larger set of error codes.

### CODE REVIEW

File: docs/config.rst [hunk 1]
```diff
@@ -415,6 +415,37 @@ Features
 
    .. versionadded:: 2.0
 
+.. attribute:: LP_ENABLE_ERROR_MEANING
```
Comment (Rycieos) [inline docs/config.rst:418]: This config variable is never checked, so the feature is always enabled.

File: docs/config.rst [hunk 2]
```diff
@@ -415,6 +415,37 @@ Features
...
+
+   Extends the set of interpreted error codes to a larger set of (POSIX) codes.
+
+   .. note:: This use a reasonable set of error codes
+             that are common on POSIX systems on x86 or ARM architectures
+             (most notably from ``sysexit.h`` and ``signal.h``).
+             But any software may use its own set of codes,
+             and thus the guess may be wrong.
+
+   See also: :attr:`LP_COLOR_ERR`.
```
Comment (Rycieos) [inline docs/config.rst:445]: This need a mention that it will have no effect if `LP_ENABLE_ERROR_MEANING` is disabled.

File: liquidprompt [hunk 3]
```diff
@@ -1258,6 +1261,200 @@ _lp_error_color() {
     lp_error_color="${LP_COLOR_ERR}${lp_error}${NO_COL}"
 }
 
+_lp_error_meaning() {
+    # Reasonable codes for a POSIX env.
+
+    _lp_error || return "$?"
```
Comment (Rycieos) [inline liquidprompt:1267]: The issue with checking for the error code in this way is that if `LP_ENABLE_ERROR` is false, this function will be disabled as well. I think it makes sense to allow users to enable error meanings without enabling the printing of the raw code.

I would change this check to this:
```suggestion
    (( LP_ENABLE_ERROR_MEANING )) || return 2
    (( lp_error != 0 )) || return 1
```

**Review by Rycieos — CHANGES_REQUESTED** [review]
Looks good in general, just a few issues.

**Comment by nojhan** [comment]
Do I squash all commits into one ? (i.e. do you want to keep the extended error meaning setup?)

**Review by Rycieos — CHANGES_REQUESTED** [review]
Looks good now.

I missed that question the first time, sorry. I do not have much of an opinion. Clearly you intended for both bits to be part of the feature, so it would be fine in once commit. But if you want it as separate, I'm fine with that.

But no matter what, squash away the extra commits from the fixes and docs.

**Comment by nojhan** [comment]
The two commits were only there to make it easy to just have a single config, if you're OK with the "extended" config, I'll squash everything.

File: liquidprompt [hunk 4]
```diff
@@ -1258,6 +1261,202 @@ _lp_error_color() {
     lp_error_color="${LP_COLOR_ERR}${lp_error}${NO_COL}"
 }
 
+_lp_error_meaning() {
+    # Reasonable codes for a POSIX env.
+
+    (( LP_ENABLE_ERROR_MEANING )) || return 2
+
+    (( _lp_error != 0 )) || return 1
```
Comment (Rycieos) [inline liquidprompt:1269]: > ./liquidprompt: line 1331: _lp_error: unbound variable

```suggestion
    (( lp_error != 0 )) || return 1
```

**Review by Rycieos — CHANGES_REQUESTED** [review]
Yeah, squash down to 1.

**Review by Rycieos — APPROVED** [review]

**Pull request merged**